docs/vm-installation.md

```diff
@@ -0,0 +1,155 @@
+
+1. Enable `ExternalNode` feature on the `antrea-controller`.
+2. Create a Namespace for `antrea-agent`. This document will use `vm-ns` as an example NameSpace for illustration.
+
+```bash
+kubectl create ns vm-ns
+```
+
+3. Create a ServiceAccount, ClusterRole and ClusterRoleBinding for `antrea-agent` as shown below. If you use a different
+   Namespace other than `vm-ns`, you need to update the [VM RBAC manifest](../build/yamls/externalnode/vm-agent-rbac.yml)
+   and change `vm-ns` to the right Namespace.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/antrea-io/antrea/feature/externalnode/build/yamls/externalnode/vm-agent-rbac.yml
+```
+
+4. Create `antrea-agent.kubeconfig` file for `antrea-agent` to access the K8S API server.
+
+```bash
+export CLUSTER_NAME="kubernetes"
+export SERVICE_ACCOUNT="vm-agent"
+APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-cluster kubernetes --server=$APISERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-context antrea-agent@kubernetes --cluster=kubernetes --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.kubeconfig use-context antrea-agent@kubernetes
+# Copy antrea-agent.kubeconfig to VM
+```
+
+5. Create `antrea-agent.antrea.kubeconfig` file for `antrea-agent` to access the `antrea-controller` API server.
+
+```bash
+# Specify the antrea-controller API server endpoint. Antrea-Controller needs to be exposed via the Node IP or a
+# public IP that is reachable from the VM
+export ANTREA_API_SERVER="https://172.18.0.1:443"
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-cluster antrea --server=$ANTREA_API_SERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-context antrea-agent@antrea --cluster=antrea --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig use-context antrea-agent@antrea
+# Copy antrea-agent.antrea.kubeconfig to VM
+```
+
+## Installation on Linux VM
+
+### Prerequisites
+
+OVS needs to be installed on the VM. For more information about OVS installation please refer to the [getting-started guide](getting-started.md#open-vswitch).
+
+### Installation
+
+1. Build `antrea-agent` binary in the root of the antrea code tree and copy the binary to Linux VM.
+
+```bash
+make docker-bin
+```
+
```

Comment:
Add a sentence about copying the binary and antrea-agent.conf to the VM? Better to provide the source path for these files.

Reply:
Would this do?

1. Build `antrea-agent` binary in the root of the antrea code tree and copy the `antrea-agent` binary from the bin
   directory to Linux VM.

```bash
make docker-bin
```

2. Copy the `antrea-agent` [configuration file](../build/yamls/externalnode/conf/antrea-agent.conf) to VM and edit the
   `antrea-agent.conf` file to set `clientConnection`, `antreaClientConnection` and `externalNodeNamespace` with the
   correct values. It specifies agent configuration parameters.

Reply:
Sounds good to me. A few nits:

bin -> `bin`

the Linux VM

`[Prerequisites on Kubernetes cluster](#prerequisites-on-kubernetes-cluster)`